src/Patches/ERScripts.cs

```diff
@@ -395,59 +397,76 @@ public static void RandomizePortals(int seed) {
 
             TunicLogger.LogTesting("created the secondary portal lists");
 
+            bool TooFewPortalsForDirectionPairs(int direction, int offset = 0) {
+                if (twoPlusPortalDirectionTracker[direction] <= deadEndPortalDirectionTracker[directionPairs[direction]] + offset) {
+                    return false;
+                }
+                if (twoPlusPortalDirectionTracker[directionPairs[direction]] <= deadEndPortalDirectionTracker[direction] + offset) {
+                    return false;
+                }
+                return true;
+            }
+
+            bool VerifyDirectionPair(Portal portal1, Portal portal2) {
+                return portal1.Direction == directionPairs[portal2.Direction];
+            }
+
+            bool decoupledEnabled = SaveFile.GetInt(SaveFlags.Decoupled) == 1;
+            bool dirPairsEnabled = SaveFile.GetInt(SaveFlags.PortalDirectionPairs) == 1;
+
             int seedIncrement = 0;  // for when a specific part fails and needs to load a portal back in
```

Comment:
Does this never gets incremented?

Reply:
Ah right, this used to be needed, but with the rewrite it's not necessary anymore